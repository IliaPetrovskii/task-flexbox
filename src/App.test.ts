import { configureToMatchImageSnapshot } from 'jest-image-snapshot';
import puppeteer, { Page, Browser } from 'puppeteer';
import path from 'path';

const customConfig = { threshold: 0 };
const toMatchImageSnapshot = configureToMatchImageSnapshot({
    customDiffConfig: customConfig,
    noColors: true,
    failureThreshold: 0.08,
    failureThresholdType: 'percent'
});
expect.extend({ toMatchImageSnapshot });

let browser: Browser;

beforeAll(async () => {
    browser = await puppeteer.launch({
        headless: true,
    });
});

afterAll(async () => {
    await browser.close();
});

describe('Flexbox', () => {
    let page: Page;
    beforeEach(async () => {
        page = await browser.newPage();
        await page.goto(`file:${path.join(__dirname, 'index.html')}`);
    });

    it('Верстка не тронута', async () => {
        const container = await page.evaluate(() => document.body.innerHTML);

        expect(container).toMatchSnapshot();
    });

    it('Рисуется нужное изображение', async () => {
        const image = await page.screenshot();

        expect(image).toMatchImageSnapshot();
    });
});
