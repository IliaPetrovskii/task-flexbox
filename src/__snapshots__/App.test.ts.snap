// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flexbox Верстка не тронута 1`] = `
"
        <section class=\\"flex-container\\">
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
        </section>
    

"
`;
